drake/solvers/MultipleTreeProblem.m

```diff
@@ -0,0 +1,463 @@
+      nArgs = length(varargin);
+      if round(nArgs/2)~=nArgs/2
+        error('Needs propertyName/propertyValue pairs')
+      end
+      for pair = reshape(varargin,2,[])
+        inpName = lower(pair{1});
+        if any(strcmp(inpName,optNames))
+          opt.(inpName) = pair{2};
+        else
+          error('%s is not a recognized parameter name',inpName)
+        end
+      end
+      
+      obj.robot = robot;
+      obj.endEffectorId = endEffectorId;
+      obj.xStart = xStart;
+      obj.xGoal = xGoal;
+      obj.nTrees = size(xStartAddTrees, 2) + 2;
+      obj.goalConstraints = goalConstraints;
+      obj.additionalConstraints = additionalConstraints;
+      obj.qNom = qNom;
+      obj.mergingThreshold = opt.mergingthreshold;
+      obj.capabilityMap = opt.capabilitymap;
+      obj.graspingHand = opt.graspinghand;
+      obj.minDistance = opt.mindistance;
+      obj.activeCollisionOptions = opt.activecollisionoptions;
+      obj.endEffectorPoint = opt.endeffectorpoint;
+      obj.status = obj.EXPLORING;
+      
+      %compute sampling space
+      xyzMin = [min([xStart(1:2), xGoal(1:2)], [], 2) - 0.5; 0];
+      xyzMax = [max([xStart(1:2), xGoal(1:2)], [], 2) + 0.5; max([xStart(3), xGoal(3)]) + 1];
+      
+      %initialize trees
+      obj.trees = OptimalMotionPlanningTree.empty(obj.nTrees, 0);  
+      for t = 1:obj.nTrees
+        obj.trees(t) = OptimalMotionPlanningTree(obj.robot, obj.endEffectorId, obj.endEffectorPoint);
+        obj.trees(t).steerFactor = opt.steerfactor;
+        obj.trees(t) = obj.trees(t).setMinDistance(obj.minDistance);
+        obj.trees(t) = obj.trees(t).setOrientationWeight(opt.orientationweight);
+        obj.trees(t).max_edge_length = opt.maxedgelength;
+        obj.trees(t).max_length_between_constraint_checks = opt.maxedgelength;
+        obj.trees(t).angle_tol = opt.angletol;
+        obj.trees(t).position_tol = opt.positiontol;
+        obj.trees(t).trees{obj.trees(t).cspace_idx}.active_collision_options = opt.activecollisionoptions;
+        obj.trees(t).trees{obj.trees(t).cspace_idx}.ikoptions = opt.ikoptions;
+        obj.trees(t) = obj.trees(t).setTranslationSamplingBounds(xyzMin, xyzMax);
+        obj.trees(t) = obj.trees(t).addKinematicConstraint(obj.additionalConstraints{:});
+        obj.trees(t) = obj.trees(t).setNominalConfiguration(obj.qNom);
+        obj.trees(t) = obj.trees(t).compile();
+        if t == 1
+          obj.trees(t) = obj.trees(t).setLCMGL('Tree 1 (Start Pose)',[1,0,0]);
+        elseif t == 2
+          obj.trees(t) = obj.trees(t).setLCMGL('Tree 2 (Goal Pose)',[0,0,1]);
+        else
+          obj.trees(t) = obj.trees(t).setLCMGL(sprintf('Tree %d', t),[1,0,1]);
+        end
+      end
+      assert(obj.trees(1).isValid(xStart))
+      
+      %set tree starting points
+      obj.startPoints = [obj.xStart, zeros(obj.robot.num_positions + 7, 1), xStartAddTrees];
+    end
+    
+    function [obj, info, cost, qPath] = rrt(obj, options)
```

Comment:
It's a bit confusing that a method named `rrt` calls both `rrtIteration` and `rrtStarIteration`. Could this be renamed?